.github/workflows/dev-email-worker.yml

```diff
@@ -11,25 +11,77 @@ on:
     branches:
       - main
 
-jobs:
+jobs:  
   unit-test:
     uses: SierraSystems/reusable-workflows/.github/workflows/java-unit-tests.yml@main
     with:
       working_directory: "src"
       profile: dps-email-worker
     secrets:
       nexus_url: ${{ secrets.NEXUS_URL }}
-
+   
   app-version:
-    uses: SierraSystems/reusable-workflows/.github/workflows/java-maven-app-version.yml@main
-    with:
-      working_directory: "src/dps-email-worker"
+    name: Get the app-version from the POM file
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: "src/dps-email-worker"
+  
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Set up JDK 8"
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: 8
+  
+      # Runs build steps
+      - name: Maven Package
+        run: mvn -ntp clean package
+        
+      - name: Maven Verify
+        run: mvn -ntp clean verify
 
-  build-image:
+  
+# This job is used to build & scan image using trivy vulnerability scanner and upload the results to github security tab
+  build-scan-image:
     needs:
       - unit-test
       - app-version
-    uses: SierraSystems/reusable-workflows/.github/workflows/docker-build-image.yml@main
-    with:
-      app_name: "dps-email-worker"
-      working_directory: .
+    name: Build & Scan Image
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: "."
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        
+      #Build image for scanning
+      - name: Build an image from Dockerfile
+        env:
+          COMPOSE_DOCKER_CLI_BUILD: 1
+          DOCKER_BUILDKIT: 1
+        run: |
+          docker build -t dev-email-worker:${{ github.sha }} .
```

Comment:
I don't see a problem with using the sha, but why are you using the sha rather than just a tag? Are you planning on using a cache in the near future?